src/main/java/com/twilio/oai/TwilioPhpGenerator.java

```diff
@@ -1,40 +1,89 @@
+            additionalProperties,
+            new ResourceMap(new Inflector()),
+            new PhpCaseResolver());
+    private final List<CodegenModel> allModels = new ArrayList<>();
+    private final Map<String, String> modelFormatMap = new HashMap<>();
+    private IAPIResourceBuilder phpAPIResource = null;
+    private IConventionMapper conventionMapper =  new LanguageConventionResolver(PHP_CONVENTIONAL_MAP_PATH);
 
     public TwilioPhpGenerator() {
         super();
-
         twilioCodegen = new TwilioCodegenAdapter(this, getName());
     }
 
+    @Override
+    public String toApiFilename(final String name) {
+        return directoryStructureService.toApiFilename(super.toApiFilename(name));
+    }
+
     @Override
     public void processOpts() {
         super.processOpts();
-
-        apiTemplateFiles.clear();
-        apiTestTemplateFiles.clear();
-        modelTestTemplateFiles.clear();
-        apiDocTemplateFiles.clear();
-        modelDocTemplateFiles.clear();
         twilioCodegen.processOpts();
     }
 
     @Override
-    public Map<String, ModelsMap> postProcessAllModels(Map<String, ModelsMap> objs) {
+    public void processOpenAPI(final OpenAPI openAPI) {
+        final String domain = twilioCodegen.getDomainFromOpenAPI(openAPI);
+        twilioCodegen.setDomain(domain);
+        directoryStructureService.configure(openAPI);
+    }
+
+    @Override
+    public Map<String, ModelsMap> postProcessAllModels(Map<String, ModelsMap> allModels) {
+        final Map<String, ModelsMap> results = super.postProcessAllModels(allModels);
+
+        Utility.addModelsToLocalModelList(results, this.allModels);
+        Utility.setComplexDataMapping(this.allModels, this.modelFormatMap);
+
+        // Return an empty collection so no model files get generated.
         return new HashMap<>();
     }
 
+    @Override
+    public OperationsMap postProcessOperationsWithModels(final OperationsMap objs, List<ModelMap> allModels) {
+        final OperationsMap results = super.postProcessOperationsWithModels(objs, allModels);
+        final List<CodegenOperation> opList = directoryStructureService.processOperations(results);
+        APIResources apiResources = processCodegenOperations(opList);
+        String resource = apiResources.getApiName();
+        results.put("resources", apiResources);
+        return results;
+    }
 
     @Override
     public String getName() {
-        return "twilio-php";
+        return TwilioHelperSDK.TWILIO_PHP.getName();
+    }
+
+    private APIResources processCodegenOperations(List<CodegenOperation> opList) {
+         return new PHPAPIResourceBuilder(new PHPAPITemplate(this), opList, this.allModels)
+                 .setAdditionalProps(directoryStructureService)
+                 .setTemplate()
+                 .setOperations(new PhpParameterResolver(conventionMapper))
+                 .setResponseModel(new PhpPropertyResolver(conventionMapper))
```

Comment:
These function names are not intuitive. I'd expected `setOperations` to take `operations` as an arg and apply it to the target object.